src/main/java/javaslang/collection/Stream.java

```diff
@@ -793,7 +793,11 @@ public Short next() {
 
     @Override
     default Stream<T> dropRight(int n) {
-        return reverse().drop(n).reverse();
+        if(n <= 0) {
+            return this;
```

Comment:
Just saw that length is Lazy. When we do it elsewhere, e.g. in Stream.takeRight we should also check `n >= length()` here.